shardy/dialect/sdy/ir/attrs.td

```diff
@@ -431,6 +450,15 @@ def Sdy_TensorSharding : AttrDef<Sdy_Dialect, "TensorSharding"> {
 
     The mesh this sharding is bound to can either be specified by a symbol
     name, referencing a corresponding `MeshOp` symbol, or an inlined `MeshAttr`.
+
+    **Constraints:**
+    - Elements in `replicated_axes` must satisfy the constraints listed in `AxisRefListAttr`.
+    - If `type` isn't a `ShapedType`, the sharding must have rank 0 and no replicated axes.
```

Comment:
In this case, because of the description, I think saying corresponding tensor type should be clear enough. In other cases we may need to look at the description as well. I'll try to review the other cases and see if I can improve on them.

Reply:
thanks!